### ISSUE TITLE
Conditional $merge and $patch keywords?

### ISSUE THREAD

**Issue by manuschillerdev** [epoberezkin/ajv-merge-patch#46]
I am trying to `merge/patch` schemas based on conditions with `if/then/else`. Is this generally possible? 

Here is a minimal reproduction CSB of what I am trying to achieve:
https://codesandbox.io/s/serene-cdn-0ugtt?file=/src/index.js

**Comment by epoberezkin** [comment]
Looking at the example, the full path that you are trying to use in patch should be `/if/properties/q`, not `/properties/q` - did you try that?

Also, it may be just a contrived example, but why do you need to repeat the keywords from `if` in `then`? If it is validating agains `then`, the condition in `if` is definitely satisfied.

Also, you might need `else: false` in case you want validation to fail unless condition in `if` is valid...

**Comment by epoberezkin** [comment]
No, was a bit confused... I am not quite sure what you are trying to achieve - other than some strict mode requirements (you can either add missing `type` keywords or use the option strict: false) it seems to be working?
